src/socket/tcp.rs

```diff
@@ -1488,6 +1488,7 @@ impl<'a> fmt::Write for TcpSocket<'a> {
 }
 
 #[cfg(test)]
+#[cfg(feature = "proto-ipv4")]
```

Comment:
Let's add `pub(crate) const MOCK_IP_ADDRESS_1` in `smoltcp::wire::ip::test` or something like this, it's not good to drop tests in IPv6...

Reply:
After looking at the tests, I'm not sure how a set of mock IP addresses would work. So I added the `ipv4` specific stuff in a `ipv4_locals` module. Would it be an public mock `IpAddress` be an `IPv6` variant if `proto-ipv6` was enabled and `IPv4` if not? Or would we have a public collection of mock IPv4 addresses and a public collection of IPv6 addresses.